### PR TITLE
Implement dynamo bucket4j pruner

### PR DESCRIPTION
**Pull request by tgregory-block**
Implements a stale bucket pruner for DynamoDb.

### CODE REVIEW

File: misk-rate-limiting-bucket4j-dynamodb-v1/src/main/kotlin/misk/ratelimiting/bucket4j/dynamodb/v1/DynamoDbV1Bucket4jRateLimiterModule.kt [hunk 1]
```diff
@@ -16,8 +17,9 @@ import java.time.Clock
 /**
  * Configures a [RateLimiter] that uses DynamoDb as a backend.
  */
-class DynamoDbV1Bucket4jRateLimiterModule(
-  private val tableName: String
+class DynamoDbV1Bucket4jRateLimiterModule @JvmOverloads constructor(
+  private val tableName: String,
+  private val prunerPageSize: Int = 1000
```
Comment (frojasg) [inline misk-rate-limiting-bucket4j-dynamodb-v1/src/main/kotlin/misk/ratelimiting/bucket4j/dynamodb/v1/DynamoDbV1Bucket4jRateLimiterModule.kt:22]: Could you add some docs around the tradeoff of making `prunerPageSize` bigger or smaller?

**Review by adrw — APPROVED** [review]

File: misk-rate-limiting-bucket4j-dynamodb-v1/src/main/kotlin/misk/ratelimiting/bucket4j/dynamodb/v1/DynamoDbV1BucketPruner.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import com.amazonaws.services.dynamodbv2.model.ReturnValue
+import io.micrometer.core.instrument.MeterRegistry
+import wisp.logging.getLogger
+import wisp.ratelimiting.RateLimitPrunerMetrics
+import wisp.ratelimiting.bucket4j.Bucket4jPruner
+import wisp.ratelimiting.bucket4j.ClockTimeMeter
+import java.time.Clock
+import kotlin.system.measureTimeMillis
+
+class DynamoDbV1BucketPruner @JvmOverloads constructor(
```
Comment (frojasg) [inline misk-rate-limiting-bucket4j-dynamodb-v1/src/main/kotlin/misk/ratelimiting/bucket4j/dynamodb/v1/DynamoDbV1BucketPruner.kt:17]: who's calling this?

Comment (tgregory-block) [reply misk-rate-limiting-bucket4j-dynamodb-v1/src/main/kotlin/misk/ratelimiting/bucket4j/dynamodb/v1/DynamoDbV1BucketPruner.kt:17]: Currently nobody, it needs to be implemented in services via some recurring task mechanism, e.g. misk-cron

File: misk-rate-limiting-bucket4j-dynamodb-v1/src/main/kotlin/misk/ratelimiting/bucket4j/dynamodb/v1/DynamoDbV1Bucket4jRateLimiterModule.kt (hunk 1 above)
Comment (tgregory-block) [reply misk-rate-limiting-bucket4j-dynamodb-v1/src/main/kotlin/misk/ratelimiting/bucket4j/dynamodb/v1/DynamoDbV1Bucket4jRateLimiterModule.kt:22]: Can do 👍

**Pull request merged**